File: bitwise_problems/number_of_1_bits/solution.js
// time complexity---O(no of digits)
/**
 * @param {number} n
 * @return {number}
 */
var hammingWeight = function(num) {
    var count=0;
        while(num>0)
            {
                last=num&1;
                if(last==1)
                    {
                        count+=1;
                    }
                num=num>>1;

            }
            return count;

    
};

// time complexity---O(no of set bits)
/**
 * @param {number} n
 * @return {number}
 */
var hammingWeight = function(num) {
    var count=0;
        while(num>0)
            {
                num=num&(num-1);
                count+=1

            }
        return count;
    
};

File: bitwise_problems/number_of_1_bits/solution.py
'''
this problem can be oslved by bitwise operators--O(log(digits))
'''
class Solution:
    def hammingWeight(self, n: int) -> int:
        count=0
        while n>0:
            last=n&1
            if last==1:
                count+=1
            n=n>>1
        return count
        
'''
 time complexity----O(number of 1 bits)
 '''       
class Solution:
    def hammingWeight(self, n: int) -> int:
        count=0
        while n>0:
            n=n&(n-1)
            count+=1
        return count